src/services/directory.ts

```diff
@@ -2,6 +2,29 @@ import { Member } from '../types';
 import api from './api';
 import { devPrint } from '../components/utils/RandomUtils';
 
+function deserializeMember(data: any): Member {
```

Comment:
I'm not sure what's going on here, you're just renaming user to id, and first_name to firstName. Most of the fields are the same. Why are we doing this?

Reply:
Our domain objects don't necessarily need to map 1:1 from frontend to backend, and likely won't as we add functionality to either end. In this case the only difference is camel vs snake case, but if we make any additional changes to api layer or frontend, it's nice to only need to update it in one place (the deserializer) as opposed to everywhere we make an api call. Any alternative patterns you suggest using to accomplish this?

Reply:
to be consistent why not just return the data how you want it with the api call? Create a function called `getMember()` or something that does the api call and returns the data how you want it. 

It's probably better to create functions that make the calls anyways. `getMember()` returns a user object or throws an error.

Reply:
Is this not exactly what the functions in the file are doing? The problem is that multiple functions will inevitably return the same type of object, and so if and when we change what's returned from the backend we will need to then update every function that returns a member as opposed to just updating the serializer.

Reply:
I see, yeah that's fair. I didn't look that closely right now before making that comment so that's perfect.